File: testdata/src/ping/ping.go
// +build ignore

package main

import (
	"fmt"

	_ "v.io/core/veyron/profiles"
	"v.io/core/veyron2"
	"v.io/core/veyron2/vlog"

	"pingpong"
)

func main() {
	ctx, shutdown := veyron2.Init()
	defer shutdown()

	s := pingpong.PingPongClient("pingpong")

	fmt.Printf("Pinging\n")
	pong, err := s.Ping(ctx, "PING")
	if err != nil {
		vlog.Fatal("error pinging: ", err)
	}
	fmt.Println(pong)
}

File: testdata/src/ping/ping.js
var veyron = require('veyron');

veyron.init(function(err, rt) {
  if (err) throw err;

  var ctx = rt.getContext();

  rt.newClient().bindTo(ctx, 'pingpong', function(err, s) {
    if (err) throw err;

    s.ping(ctx, 'PING', function(err, pong) {
      if (err) throw err;

      console.log(pong);
      process.exit(0);
    });
  });
});
